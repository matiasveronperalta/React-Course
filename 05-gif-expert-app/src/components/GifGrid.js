import React from 'react'

export const GifGrid = ({category}) => {

    const getGifs = async() =>{
        const url = 'https://api.giphy.com/v1/gifs/search?q=rick+and+morty&limit=10&api_key=UkXtWw1t02SZK27KypfrzFJ3wf5sWLnz';
        const response = await fetch(url);
        const {data} = await response.json();

        const gifs = data.map( img => {
            return{
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url,
            }
        });

        console.log(gifs);
    }

    getGifs();

    return (
        <>
            <h3>{category}</h3>
        </>
    )
}
