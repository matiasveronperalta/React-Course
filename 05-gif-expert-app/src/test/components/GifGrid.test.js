import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../Hooks/useFetchGifs';
jest.mock('../../Hooks/useFetchGifs');


describe('Test <GifGrid/>', () => {

    test('should be show correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category="Dragon Ball" />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should be show item where load image useFetchGifs', () => {

        const images = [{
            id:'ABC',
            url:'https://localhost:80',
            title: 'Nombre'
        }];

        useFetchGifs.mockReturnValue({
            data: images,
            loading: false
        });

        const wrapper = shallow(<GifGrid category="Dragon Ball" />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(images.length);

    })
});


