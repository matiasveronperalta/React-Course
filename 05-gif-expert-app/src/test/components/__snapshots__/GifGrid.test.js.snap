// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid/> should be show correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test <GifGrid/> should be show item where load image useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Nombre"
      url="https://localhost:80"
    />
  </div>
</Fragment>
`;
